logstash-core/src/main/java/org/logstash/ackedqueue/ext/JrubyAckedQueueExtLibrary.java

```diff
@@ -128,7 +131,8 @@ public IRubyObject ruby_open(ThreadContext context)
             try {
                 this.queue.open();
             } catch (IOException e) {
-                throw context.runtime.newIOErrorFromException(e);
+                // will preserve Java stacktrace & bubble up as a Ruby IOError
+                throw new RaiseException(e, new NativeException(context.runtime, context.runtime.getIOError(), e));
```

Comment:
@colinsurprenant maybe extract the comment and ```throw new RaiseException(e, new NativeException(``` to a `public static` on `RubyUtil`? (No need to pollute any hot path methods by two additional constructors and potentially mess up inlining anywhere :)).

We could just add `public static void rubyThrow(Ruby runtime, Throwable e) {throw new RaiseException(e, new NativeException(runtime, runtime.getIOError(), e));}` to `RubyUtil` right?

Reply:
can you explain «pollute any hot path methods by two additional constructors and potentially mess up inlining anywhere» especially the "pollute any hot path methods" and "mess up inlining anywhere" ?

Reply:
@colinsurprenant sure:

* The JVM inlines methods of up to a certain size (also depending on how often they are called) into callers
   * Some of these methods are called very often i.e. are hot, the exception throwing part is almost never called though (I guess ideally never :))
* If you have logic for throwing these exceptions inline in the method, this will increase the size of the method needlessly => make it less likely to be inlined into callers
* If that logic is in a separate method that never gets called, then only the method call will count towards the byte code size of the method making it smaller and more likely to be inlined

This is kind of a nice write-up of the problem I think:
http://normanmaurer.me/blog/2014/05/15/Inline-all-the-Things/

Reply:
yes, sorry, I should have more precisely said: especially the "pollute any hot path methods" and "mess up inlining anywhere in the context of a catch block which, as you say, is ideally never called. 
while I do understand this, I believe the practical benefit for doing this change is certainly near zero. 
but sure, there is no downside in creating a method for it and it will be more concise.
thanks for the explanation.